op-chain-ops/interopgen/deploy.go

```diff
@@ -0,0 +1,338 @@
+	// after creating the final config for any particular L2. Will add comments.
+
+	for l2ChainID, l2Cfg := range cfg.L2s {
+		l2Deployment, err := deployL2ToL1(l1Host, cfg.Superchain, superDeployment, l2Cfg)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to deploy L2 %d to L1: %w", &l2ChainID, err)
+		}
+		deployments.L2s[l2ChainID] = l2Deployment
+	}
+
+	out := &WorldOutput{
+		L2s: make(map[string]*L2Output),
+	}
+	l1Out, err := completeL1(l1Host, cfg.L1)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to complete L1: %w", err)
+	}
+	out.L1 = l1Out
+
+	// Now that the L1 does not change anymore we can compute the L1 genesis block, to anchor all the L2s to.
+	l1GenesisBlock := l1Out.Genesis.ToBlock()
+	genesisTimestamp := l1Out.Genesis.Timestamp
+
+	for l2ChainID, l2Cfg := range cfg.L2s {
+		l2Host := createL2(logger, fa, srcFS, l2Cfg, genesisTimestamp)
+		if err := l2Host.EnableCheats(); err != nil {
+			return nil, nil, fmt.Errorf("failed to enable cheats in L2 state %s: %w", l2ChainID, err)
+		}
+		if err := genesisL2(l2Host, l2Cfg, deployments.L2s[l2ChainID]); err != nil {
+			return nil, nil, fmt.Errorf("failed to apply genesis data to L2 %s: %w", l2ChainID, err)
+		}
+		l2Out, err := completeL2(l2Host, l2Cfg, l1GenesisBlock, deployments.L2s[l2ChainID])
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to complete L2 %s: %w", l2ChainID, err)
+		}
+		out.L2s[l2ChainID] = l2Out
+	}
+	return deployments, out, nil
+}
+
+func createL1(logger log.Logger, fa *foundry.ArtifactsFS, srcFS *foundry.SourceMapFS, cfg *L1Config) *script.Host {
+	l1Context := script.Context{
+		ChainID:      cfg.ChainID,
+		Sender:       sysGenesisDeployer,
+		Origin:       sysGenesisDeployer,
+		FeeRecipient: common.Address{},
+		GasLimit:     script.DefaultFoundryGasLimit,
+		BlockNum:     uint64(cfg.L1GenesisBlockNumber),
+		Timestamp:    uint64(cfg.L1GenesisBlockTimestamp),
+		PrevRandao:   cfg.L1GenesisBlockMixHash,
+		BlobHashes:   nil,
+	}
+	l1Host := script.NewHost(logger.New("role", "l1", "chain", cfg.ChainID), fa, srcFS, l1Context)
+	return l1Host
+}
+
+func createL2(logger log.Logger, fa *foundry.ArtifactsFS, srcFS *foundry.SourceMapFS, l2Cfg *L2Config, genesisTimestamp uint64) *script.Host {
+	l2Context := script.Context{
+		ChainID:      new(big.Int).SetUint64(l2Cfg.L2ChainID),
+		Sender:       sysGenesisDeployer,
+		Origin:       sysGenesisDeployer,
+		FeeRecipient: common.Address{},
+		GasLimit:     script.DefaultFoundryGasLimit,
+		BlockNum:     uint64(l2Cfg.L2GenesisBlockNumber),
+		Timestamp:    genesisTimestamp,
+		PrevRandao:   l2Cfg.L2GenesisBlockMixHash,
+		BlobHashes:   nil,
+	}
+	l2Host := script.NewHost(logger.New("role", "l2", "chain", l2Cfg.L2ChainID), fa, srcFS, l2Context)
+	l2Host.SetEnvVar("OUTPUT_MODE", "none") // we don't use the cheatcode, but capture the state outside of EVM execution
+	l2Host.SetEnvVar("FORK", "granite")     // latest fork
```

Comment:
This will be a maintenance burden, is there a way to use the chain config to determine which fork to use? If not its fine for now, I do think that this duplicates work with `op-deploy` a bunch which is fine to ship fast but we should try to consolidate into a single tool as much as possible, I know that @mslipper was into the idea of having multiple backends for `op-deploy`, one could just output JSON rather than send transactions

Reply:
Yes, I've reached out to @mslipper and I am trying to coordinate with him now how we merge interopgen and op-deployer work